### PR TITLE
enhance ConfigureMake generic easyblock to add support for building multiple build targets

### PR DESCRIPTION
**Pull request by mboisson**

### CODE REVIEW

**Comment by mboisson** [comment]
Here is an example of EasyConfig which would use this feature: 
https://github.com/ComputeCanada/easybuild-easyconfigs/blob/computecanada-main/easybuild/easyconfigs/a/Arioc/Arioc-1.43-gcccorecuda-2020a.eb

File: easybuild/easyblocks/generic/configuremake.py [hunk 1]
```diff
@@ -175,6 +176,7 @@ def extra_options(extra_vars=None):
         extra_vars = EasyBlock.extra_options(extra=extra_vars)
         extra_vars.update({
             'build_cmd': [DEFAULT_BUILD_CMD, "Build command to use", CUSTOM],
+            'build_targets': [DEFAULT_BUILD_TARGETS, "Targets to build", CUSTOM],
```
Comment (boegel) [inline easybuild/easyblocks/generic/configuremake.py:179]: This should say "`Target name (string) or list of target names to build`"?

File: easybuild/easyblocks/generic/configuremake.py [hunk 2]
```diff
@@ -65,6 +65,7 @@
 DEFAULT_CONFIGURE_CMD = './configure'
 DEFAULT_BUILD_CMD = 'make'
 DEFAULT_INSTALL_CMD = 'make install'
+DEFAULT_BUILD_TARGETS = ''
```
Comment (boegel) [inline easybuild/easyblocks/generic/configuremake.py:68]: `DEFAULT_BUILD_TARGET`

File: easybuild/easyblocks/generic/configuremake.py [hunk 3]
```diff
@@ -320,14 +322,20 @@ def build_step(self, verbose=False, path=None):
...
-            self.cfg.get('build_cmd') or DEFAULT_BUILD_CMD,
-            paracmd,
-            self.cfg['buildopts'],
-        ])
+        targets = self.cfg['build_targets'] or DEFAULT_BUILD_TARGETS
+        # ensure it is a list
+        targets = [targets] if isinstance(targets, str) else targets
+
+        for target in targets:
+            cmd = ' '.join([
```
Comment (boegel) [inline easybuild/easyblocks/generic/configuremake.py:330]: let's add a log message here:

```python
self.log.info("Building target '%s'", target)

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/generic/configuremake.py (hunk 1 above)
Comment (mboisson) [reply easybuild/easyblocks/generic/configuremake.py]: done

**Review by smoors — APPROVED** [review]
tested with the easyconfig linked to (using gcccuda), works nicely

**Comment by smoors** [comment]
Going in, thanks @mboisson!

**Pull request merged**